// defekt@9.3.0/build/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.value = exports.ResultDoesNotContainError = exports.isResult = exports.isError = exports.isCustomError = exports.hydrateResult = exports.HydratingErrorFailed = exports.hydrateCustomError = exports.error = exports.ensureUnknownIsError = exports.defekt = exports.CustomError = void 0;
const CustomError_1 = require("./CustomError");
Object.defineProperty(exports, "CustomError", { enumerable: true, get: function () { return CustomError_1.CustomError; } });
const defekt_1 = require("./defekt");
Object.defineProperty(exports, "defekt", { enumerable: true, get: function () { return defekt_1.defekt; } });
const ensureUnknownIsError_1 = require("./ensureUnknownIsError");
Object.defineProperty(exports, "ensureUnknownIsError", { enumerable: true, get: function () { return ensureUnknownIsError_1.ensureUnknownIsError; } });
const hydrateResult_1 = require("./hydrateResult");
Object.defineProperty(exports, "hydrateResult", { enumerable: true, get: function () { return hydrateResult_1.hydrateResult; } });
const isCustomError_1 = require("./isCustomError");
Object.defineProperty(exports, "isCustomError", { enumerable: true, get: function () { return isCustomError_1.isCustomError; } });
const isError_1 = require("./isError");
Object.defineProperty(exports, "isError", { enumerable: true, get: function () { return isError_1.isError; } });
const isResult_1 = require("./isResult");
Object.defineProperty(exports, "isResult", { enumerable: true, get: function () { return isResult_1.isResult; } });
const Result_1 = require("./Result");
Object.defineProperty(exports, "error", { enumerable: true, get: function () { return Result_1.error; } });
Object.defineProperty(exports, "ResultDoesNotContainError", { enumerable: true, get: function () { return Result_1.ResultDoesNotContainError; } });
Object.defineProperty(exports, "value", { enumerable: true, get: function () { return Result_1.value; } });
const hydrateCustomError_1 = require("./hydrateCustomError");
Object.defineProperty(exports, "hydrateCustomError", { enumerable: true, get: function () { return hydrateCustomError_1.hydrateCustomError; } });
Object.defineProperty(exports, "HydratingErrorFailed", { enumerable: true, get: function () { return hydrateCustomError_1.HydratingErrorFailed; } });
