// defekt@9.3.0/build/lib/Result.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.error = exports.value = exports.ResultDoesNotContainError = void 0;
const defekt_1 = require("./defekt");
class ResultDoesNotContainError extends (0, defekt_1.defekt)({ code: 'ResultDoesNotContainError' }) {
}
exports.ResultDoesNotContainError = ResultDoesNotContainError;
const error = function (err) {
    return {
        hasError() {
            return true;
        },
        hasValue() {
            return false;
        },
        unwrapOrThrow(errorTransformer) {
            if (errorTransformer) {
                throw errorTransformer(err);
            }
            // eslint-disable-next-line @typescript-eslint/no-throw-literal
            throw err;
        },
        unwrapOrElse(handleError) {
            return handleError(err);
        },
        unwrapOrDefault(defaultValue) {
            return defaultValue;
        },
        unwrapErrorOrThrow() {
            return err;
        },
        error: err
    };
};
exports.error = error;
const value = function (val) {
    return {
        hasError() {
            return false;
        },
        hasValue() {
            return true;
        },
        unwrapOrThrow() {
            return val;
        },
        unwrapOrElse() {
            return val;
        },
        unwrapOrDefault() {
            return val;
        },
        unwrapErrorOrThrow() {
            throw new ResultDoesNotContainError();
        },
        value: val
    };
};
exports.value = value;
